### PR TITLE
ci: enable DRA for main and branches

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

* Support `DRA` within the current CI pipeline so artifacts are enrolled to the Unified Release process.
* Artifacts are published to a Google bucket then DRA can download them to run the release manager helper step.

## Why is it important?

Automate the release as requested

## Issues

Similar implementation done for the `fleet-server`, though in this case there is a monolithically pipeline instead two

## Actions

- [x] Required to build a release and snapshot
- [x] Define where the binaries are stored
- [x] Who to be notified 
- [x] Get the current version

## Test

After changing the pipeline to work for PRs (only for testing purposes):

image

It worked like a charm, so artifacts are now populated to the DRA:

- https://artifacts-snapshot.elastic.co/elastic-agent-shipper/8.5.0-31576ff0/summary-8.5.0-SNAPSHOT.html
- https://artifacts-staging.elastic.co/elastic-agent-shipper/8.5.0-48bb8ecc/summary-8.5.0.html

### CODE REVIEW

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:42]: TBC, what's the file or command to be executed so the version of the project can be saved in the env varaible to be used later on

Comment (narph) [reply .ci/Jenkinsfile]: I've added https://github.com/elastic/elastic-agent-shipper/pull/102 to retrieve the version

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:14]: Who to be notified by email/slack?

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:233]: This is the project to be added to the DRA in the Unified Release process, so once it's available then we could test this particular PR (for such we might need to remove some when statements)

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:114]: Artifacts are now available in the google bucket

image

File: .ci/Jenkinsfile
Comment (cmacknz) [reply .ci/Jenkinsfile:14]: Yes, probably #elastic-agent and the elastic-agent mailing list.

**Comment by v1v** [comment]
@mergifyio update

**Comment by cmacknz** [comment]
There is some related discussion of how to integrate the shipper with the agent build in https://github.com/elastic/release-eng/issues/268

I'm not sure if it changes anything here, as a DRA based solution is the path forward. I think the only question is whether the referenced DRA phase 2 project makes it easier to consume from the perspective of the agent itself.

File: .ci/Jenkinsfile
Comment (michalpristas) [inline .ci/Jenkinsfile:13]: shipper is data plane so either `elastic-agent-data-plane` or `ingest-notifications` could be fine
@jlind23 wdyt on using `ingest-notifications`?

File: .ci/Jenkinsfile
Comment (michalpristas) [reply .ci/Jenkinsfile:114]: where can I find these? I'd like to see result of rekicked run

File: .ci/Jenkinsfile
Comment (jlind23) [inline .ci/Jenkinsfile:13]:
```suggestion
    SLACK_CHANNEL = '#ingest-notifications'
```

**Review by jlind23 — CHANGES_REQUESTED** [review]

File: .ci/Jenkinsfile
Comment (michalpristas) [reply .ci/Jenkinsfile:14]: what's this? is this a valid google group? i'd rather avoid using elastic-agent mailing list to use here as it should be product specific. i'd rather use team specific or create a new one unique for shipper

File: .ci/Jenkinsfile
Comment (v1v) [reply .ci/Jenkinsfile:114]: Those artifacts are available in the Google bucket, as we have moveod this project to `fleet-ci` is not anymore available in the `beats-ci` but `fleet-ci` google bucket

```bash
gsutil list gs://fleet-ci-artifacts/jobs/elastic-agent-shipper/commits
```

**Comment by v1v** [comment]
@mergifyio rebase

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:41]: Once the release manager supports the DRA for this project, then we can set to `true` this, to validate the DRA works as expected before merging this PR.

```suggestion
        setEnvVar('IS_BRANCH_AVAILABLE', 'true')
```

File: .ci/Jenkinsfile
Comment (michalpristas) [inline .ci/Jenkinsfile:232]:
```suggestion
  dir("${BASE_DIR}") {
    def mageGoal = args.type.equals('staging') ? 'release-manager-dependencies-release' : 'release-manager-dependencies-snapshot'
    withMageEnv() {
      sh(label: 'create dependencies file', script: "mage ${mageGoal}")
    }
```

File: .ci/Jenkinsfile
Comment (michalpristas) [inline .ci/Jenkinsfile:14]:
```suggestion
    NOTIFY_TO = 'elastic-agent+build-package@elastic.co'
```

File: .ci/Jenkinsfile
Comment (michalpristas) [inline .ci/Jenkinsfile:236]:
```suggestion
                   artifactsFolder: 'build/distributions',
```

File: .ci/Jenkinsfile
Comment (michalpristas) [inline .ci/Jenkinsfile:230]:
```suggestion
                        localDirectory: "${BASE_DIR}/build/distributions",
```

File: .ci/Jenkinsfile
Comment (michalpristas) [inline .ci/Jenkinsfile:219]:
```suggestion
    pathPrefix: "${BASE_DIR}/build/distributions/",
```

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:142]:
```suggestion
              cmd(label: 'Go package', script: 'mage -v package:all')
```

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:114]:
```suggestion
              cmd(label: 'Go package', script: 'mage -v package:all')
```

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:14]:
```suggestion
    SLACK_CHANNEL = '#observablt-bots'
    NOTIFY_TO = 'victor.martinez@elastic.co'
```

File: .ci/Jenkinsfile
Comment (v1v) [reply .ci/Jenkinsfile:235]: @michalpristas , this suggestion failed:


```
[2022-11-02T21:19:11.528Z] bash: dev-tools/dependencies: No such file or directory
```

see https://fleet-ci.elastic.co/blue/organizations/jenkins/elastic-agent-shipper-mbp/detail/PR-100/16/pipeline/586#step-630-log-10

any idea?

Comment (v1v) [reply .ci/Jenkinsfile:235]: https://github.com/elastic/elastic-agent-shipper/pull/100/commits/bc8a1649385e26011039e76e3604e1d8c3d09155 should help

**Comment by v1v** [comment]
@michalpristas , I think the name of the artifacts (tar.gz, zip) need to contain the version, similarly done for `fleet-server`:

image

https://fleet-ci.elastic.co/job/fleet-server/job/fleet-server-package-mbp/job/main/70/gcsObjects/

but I cannot see that in 

image

https://fleet-ci.elastic.co/job/elastic-agent-shipper-mbp/job/PR-100/16/gcsObjects/


https://artifacts-snapshot.elastic.co/fleet-server/8.6.0-ecb279e9/summary-8.6.0-SNAPSHOT.html

Can you please verify whether that's needed?

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:252]: to use BRANCH_NAME

```suggestion
                   branch: env.BRANCH_NAME,
```

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:14]:
```suggestion
    SLACK_CHANNEL = '#ingest-notifications'
    NOTIFY_TO = 'elastic-agent+build-package@elastic.co'
```

File: .ci/Jenkinsfile
Comment (v1v) [inline .ci/Jenkinsfile:43]:
```suggestion
        setEnvVar('IS_BRANCH_AVAILABLE', isBranchUnifiedReleaseAvailable(env.BRANCH_NAME))
```

**Review by michalpristas — APPROVED** [review]
LGTM @jlind23 if you could take a final look for channels and emails it would be very great

**Review by jlind23 — APPROVED** [review]

**Pull request merged**
